docs/messaging/azure-messaging.md

```diff
@@ -15,15 +15,70 @@ The Semantic Conventions for [Azure Service Bus](https://learn.microsoft.com/azu
+| [`messaging.servicebus.message.delivery_count`](../attributes-registry/messaging.md) | int | Number of deliveries that have been attempted for this message. | `2` | `Conditionally Required` [7] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`server.address`](../attributes-registry/server.md) | string | The domain name of the Service Bus namespace. [8] | `contoso.servicebus.windows.net` | `Conditionally Required` If available. | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
+| [`messaging.operation.name`](../attributes-registry/messaging.md) | string | The system-specific name of the messaging operation. [9] | `send`; `process`; `get_partition_properties` | `Recommended` [10] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | [`messaging.servicebus.message.enqueued_time`](../attributes-registry/messaging.md) | int | The UTC epoch seconds at which the message has been accepted and stored in the entity. | `1701393730` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 
-**[1]:** If delivery count is available and is bigger than 0.
+**[1]:** If a custom value is used, it MUST be of low cardinality.
+
+**[2]:** The error type SHOULD match the AMQP error code (when applicable) or a fully qualified type name of the exception that caused the error.
+
+**[3]:** If and only if the messaging operation has failed.
+
+**[4]:** Instrumentations SHOULD NOT set `messaging.batch.message_count` on spans that operate with a single message. When a messaging client library supports both batch and single-message API for the same operation, instrumentations SHOULD use `messaging.batch.message_count` for batching APIs and SHOULD NOT use it for single-message APIs.
+
+**[5]:** If the span describes an operation on a batch of messages.
+
+**[6]:** <!-- TODO: overriding the base note, workaround for https://github.com/open-telemetry/build-tools/issues/299 -->
+
+**[7]:** If delivery count is available and is bigger than 0.
+
+**[8]:** <!-- TODO: overriding the base note, workaround for https://github.com/open-telemetry/build-tools/issues/299 -->
+
+**[9]:** The operation name SHOULD match one of the following values (the normalized name of the client library method in snake_case):
+
+- sender operations: `send`, `schedule`, `cancel_scheduled`
+- transaction operations: `create_transaction`, `commit_transaction`, `rollback_transaction`
+- receiver operation: `receive`, `peek`, `receive_deferred`, `renew_message_lock`
+- settlement operations: `abandon`, `complete`, `defer`, `dead_letter`
+- session operation: `accept_session`, `get_session_state`, `set_session_state`, `renew_session_lock`
+
+**[10]:** If the operation is not sufficiently described by `messaging.operation.type`.
```

Comment:
Two questions:
1. Do you always expect `message.operation.name` to be populated for Azure?
2. Is it always possible to correlate an operation name with an operation type? For example, is it clear what operation type should be used for transaction and session operations?

Reply:
> Do you always expect `message.operation.name` to be populated for Azure?

I believe the only two operation names that match type are `receive` and `process`. In all other cases we'll populate operation name.

> Is it always possible to correlate an operation name with an operation type? For example, is it clear what operation type should be used for transaction and session operations?

Yes, that's a good question I wanted to bring up - I don't think `messaging.operation.type` should be `required` since there a a lot of messaging operations that semconv don't explicitly cover. 

So in the current version of the spec I'd have to set `messaging.operation.type` to a custom value, but ideally it should only be required for core operations like `send`, `receive`, `process`, and `settle`.
Let me create an issue.

Reply:
> > Do you always expect `message.operation.name` to be populated for Azure?
> 
> I believe the only two operation names that match type are `receive` and `process`. In all other cases we'll populate operation name.
> 
> > Is it always possible to correlate an operation name with an operation type? For example, is it clear what operation type should be used for transaction and session operations?
> 
> Yes, that's a good question I wanted to bring up - I don't think `messaging.operation.type` should be `required` since there a a lot of messaging operations that semconv don't explicitly cover.
> 
> So in the current version of the spec I'd have to set `messaging.operation.type` to a custom value, but ideally it should only be required for core operations like `send`, `receive`, `process`, and `settle`. Let me create an issue.

Created https://github.com/open-telemetry/semantic-conventions/issues/947 - let's discuss it on Thursday